pkg/controllers/provisioning/provisioner.go

```diff
@@ -372,14 +373,185 @@ func (p *Provisioner) getDaemonSetPods(ctx context.Context) ([]*v1.Pod, error) {
 	if err := p.kubeClient.List(ctx, daemonSetList); err != nil {
 		return nil, fmt.Errorf("listing daemonsets, %w", err)
 	}
+
+	limitRangeList := &v1.LimitRangeList{}
+	if err := p.kubeClient.List(ctx, limitRangeList); err != nil {
+		return nil, fmt.Errorf("listing daemonsets, %w", err)
+	}
+
+	limitRageMap := map[string][]v1.LimitRange{}
+	for _, lr := range limitRangeList.Items {
+		limitRageMap[lr.Namespace] = append(limitRageMap[lr.Namespace], lr)
+	}
+
 	var ret []*v1.Pod
 	for _, daemonSet := range daemonSetList.Items {
-		pod := &v1.Pod{Spec: daemonSet.Spec.Template.Spec}
-		ret = append(ret, pod)
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: daemonSet.Namespace},
+			Spec: *daemonSet.Spec.Template.Spec.DeepCopy()}
+		if applyLR(limitRageMap, pod) == nil {
+			ret = append(ret, pod)
```

Comment:
We should log this error if it occurs. If we don't, a pod will just silently get ignored.

Reply:
Can we log the error instead of just not appending the pod?  If not, this will be difficult to debug if it occurs for customers.